From 0ccb42a50e32056717106e46ea7b0f7059b8c72b Mon Sep 17 00:00:00 2001
From: Marek Vasut <marek.vasut@mailbox.org>
Date: Mon, 15 Sep 2025 02:49:05 +0200
Subject: [PATCH 10/10] ARM: stm32: Perform node compatible check for KS8851
 early

Check the compatible string of ethernet1 node for KS8851 very early on,
before calling uclass_get_device_by_of_path() which might initialize
the device and possibly attempt to configure MAC address into device
which is not KS8851. Doing the compatibility check early prevent this.

Upstream-Status: Backport [0f4bfee3cfe7e0786b8bf748c523c8e78eda8423]
Signed-off-by: Marek Vasut <marek.vasut@mailbox.org>
Reviewed-by: Patrice Chotard <patrice.chotard@foss.st.com>
---
 board/dhelectronics/dh_stm32mp1/board.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/board/dhelectronics/dh_stm32mp1/board.c b/board/dhelectronics/dh_stm32mp1/board.c
index 504acf47559..75db76f9dea 100644
--- a/board/dhelectronics/dh_stm32mp1/board.c
+++ b/board/dhelectronics/dh_stm32mp1/board.c
@@ -86,6 +86,9 @@ static bool dh_stm32_mac_is_in_ks8851(void)
 	if (!ofnode_valid(node))
 		return false;
 
+	if (!ofnode_device_is_compatible(node, "micrel,ks8851-mll"))
+		return false;
+
 	ret = ofnode_get_path(node, path, sizeof(path));
 	if (ret)
 		return false;
@@ -94,9 +97,6 @@ static bool dh_stm32_mac_is_in_ks8851(void)
 	if (ret)
 		return false;
 
-	if (!ofnode_device_is_compatible(node, "micrel,ks8851-mll"))
-		return false;
-
 	/*
 	 * KS8851 with EEPROM may use custom MAC from EEPROM, read
 	 * out the KS8851 CCR register to determine whether EEPROM
-- 
2.51.0

